src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+import { roundedDown } from './utility/roundedDown';
 
-export class Triangle implements Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => {};
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error("sides can't have zero length");
```

Comment:
[CODE STYLE] - The error message should be more descriptive. For example, it could indicate that the sides of a triangle must be positive numbers greater than zero.